src/gtc/dace/oir_to_dace.py

```diff
@@ -409,38 +417,37 @@ def get_k_subsets(self, node):
         read_subsets = dict()
         k_origins = dict()
         for name, offsets in collection.offsets().items():
-            k_origins[name] = -min(o[2] for o in offsets)
+            k_accesses = set(filter(None, {o[2] for o in offsets})) or {0}
+            k_origins[name] = -min(k_accesses)
         for name, offsets in collection.read_offsets().items():
             if self._axes[name][2]:
-                read_subsets[name] = "{origin}{min_off:+d}:{origin}{max_off:+d}".format(
-                    origin=k_origins[name],
-                    min_off=min(o[2] for o in offsets),
-                    max_off=max(o[2] for o in offsets) + 1,
-                )
+                min_off = min(o[2] for o in offsets) or 0
+                max_off = (max(o[2] for o in offsets) or 0) + 1
+                read_subsets[name] = f"{k_origins[name]}{min_off:+d}:{k_origins[name]}{max_off:+d}"
         for name in collection.write_fields():
             if self._axes[name][2]:
                 write_subsets[name] = "{origin}:{origin}+1".format(origin=k_origins[name])
         return read_subsets, write_subsets
```

Comment:
@gronerl could you take a look at these changes? I would essentially like `get_k_subsets` to return the equivalent of `Interval.full()` if there is a variable offset read (that is, if `any(o[2] is None for o in offsets`). I don't know what that representation looks like in the `f"{k_origins[name]}{min_off:+d}:{k_origins[name]}{max_off:+d}"` format. I'm not even sure if those are relative or absolute (they look relative to me).

Reply:
I think you almost had it, basically, Interval.full() would be `f"0:__K"` (although "__K" may not be meaningful for some of the inner graphs, I have to wrap my head around it.) Now, we may have more than `full` sometimes, namely if we read outside the K domain. My commit should consider this. Note though that for the dependency analysis, we just assume the interval is `Interval.full()` I hope that is not a problem ever, I think the cases where it could would be highly hypothetical. (Something like a variable-offset field being read outside of [start, end) would then be considered non-overlaping with the dynamicly accessed interval, and therefore not considered a dependency. Since all those cases are read only, I don't think it can be an issue. But I haven't thought this through.)  

Now, the round trip is not raising, the result compiles and a very simple validation test suite I added also does the right thing. Please double check if everything looks fine.  

Minor thing is that you often check of `bool(o[2])` to detect `None` however that also excludes `0`-offset values.